packages/react/src/utils/objectUtils.ts

```diff
@@ -0,0 +1,10 @@
+export const objectValuesEqual = <T extends { [key: string]: unknown }>(
```

Comment:
We should document all our utils and components with JSDOC. Maybe something like you mentioned in the PR?. This is a shallow and strict equality check etc.